src/pages/Apps/actions.js

```diff
@@ -35,19 +35,20 @@ export const useFetchAppList = ({ api }) => {
 
 const appCache = {};
 
-export const useFetchApp = ({ api, appId }) => {
+export const useFetchApp = ({ api, appId, location }) => {
   const [app, setApp] = useState(appCache[appId] || {});
   const [loading, setLoading] = useState(true);
   const [error, setError] = useState(null);
+  const refresh = location?.state?.app;
 
   useEffect(() => {
     let unmounted = false;
 
-    if (appCache[appId]) {
+    if (appCache[appId] && !refresh) {
       return;
     }
 
-    setLoading(true);
+    !refresh && setLoading(true); // Do not refresh when updating app object.
```

Comment:
If we don't do this, the whole application will be set in the loading state and it will create a weird UX. This is mainly used when the application is updated in the settings page. Instead of passing down `setApp` function, we simply allow children components to use `history.push({ state: { app: Date.now() }})` to trigger an app refetch, which updates the `app` object.